### ISSUE TITLE
Not able to execute the blockchain

### ISSUE THREAD

**Issue by parul-k** [satwikkansal/python_blockchain_app#32]
I have followed these steps to run project but I am not able to execute it:

**Step 1:** I have cloned the complete project from github.

**Step 2**: I am using windows 10 so I have opened command prompt and set:
SET FLASK_APP=node_server.py
flask run

After executing step 2: I see following message on the screen:
* Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)

**Step 3:**
When i am typing http://127.0.0.1:5000/ in browser it says:

**Internal Server Error**
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. 

When i checked the trace in command prompt, it says:

HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /chain (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x03C0A4A8>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
127.0.0.1 - - [25/Mar/2020 13:44:49] "[35m[1mGET / HTTP/1.1[0m" 500 -

Kind help in this regard is highly appreciated.

**Comment by satwikkansal** [comment]
Please make sure env variable is set properly (see [this](https://stackoverflow.com/questions/37826016/env-variables-not-set-while-running-minimal-flask-application/40852895#40852895))

And then make sure there's no other process running already on the same ports (5000 and 8000)

**Comment by gerardkm94** [comment]
Looks like the APP is trying to reach /chain, but with port 8000... it's quite weir. At first, I recommend you to debug and try to catch the Exception when raised. If you can't, you can try to add this at the end of the server.py file: 

`if __name__ == '__main__':
    app.run(debug=True, port=8000)`

And run into the terminal:
python run_app.py instead of flask, and let's see what happens.

Set these variables as well,

                "FLASK_APP": "app.py",
                "FLASK_ENV": "development",
                "FLASK_DEBUG": "0"

**Comment by parul-k** [comment]
> Looks like the APP is trying to reach /chain, but with port 8000... it's quite weir. At first, I recommend you to debug and try to catch the Exception when raised. If you can't, you can try to add this at the end of the server.py file:
> 
> `if __name__ == '__main__': app.run(debug=True, port=8000)`
> 
> And run into the terminal:
> python run_app.py instead of flask, and let's see what happens.
> 
> Set these variables as well,
> 
> ```
>             "FLASK_APP": "app.py",
>             "FLASK_ENV": "development",
>             "FLASK_DEBUG": "0"
> ```

Thanks for your kind response, I am able to see the following screen after running python run_app.py but its not doing anything, what about node_server.py !! Do I need to run that also?
![image](https://user-images.githubusercontent.com/60229741/77523641-0bd28e00-6eac-11ea-9fd2-c38832ec66be.png)

**Comment by gerardkm94** [comment]
Yes, my mistake. Add if __name... to node_server.py and the, python node_server.py. Is not necessary to make python run_app.py.

**Comment by parul-k** [comment]
> Yes, my mistake. Add if __name... to node_server.py and the, python node_server.py. Is not necessary to make python run_app.py.

if I am running, run_app.py with following code,
from app import app
app.run(debug=True)
it displays:
-----------
requests.exceptions.ConnectionError
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /chain (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x041A8DA8>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it'))
-------------------------
but if run the same file with,
from app import app
app.run(debug=True, port=8000)
I am able to see the following, what is happening I am not getting:
YourNet: Decentralized content sharing......

**Comment by parul-k** [comment]
Modified node_server.py file and added:
if __name__ == '__main__':
    app.run(debug=True, port=8000)
but it is displaying the following:

![image](https://user-images.githubusercontent.com/60229741/77525576-44c03200-6eaf-11ea-8dee-dbdca88c4e88.png)

**Comment by satwikkansal** [comment]
Your issue seems similar to https://github.com/satwikkansal/python_blockchain_app/issues/25

**Comment by parul-k** [comment]
> Your issue seems similar to #25

You are right but I tried everything on referred page, still it says page not found..
This is why in my case run_app.py is running fine on port 5000 but node_server.py is saying 404 Not Found..
How to check whether flask is running node_server.py as a Server or web application, any suggestions ?
This is what I am doing:
![image](https://user-images.githubusercontent.com/60229741/77538871-7264a580-6ec6-11ea-973e-a4c39cdc9fe1.png)
and result is
![image](https://user-images.githubusercontent.com/60229741/77538948-8f997400-6ec6-11ea-88bd-3e7a00ac0ab5.png)

**Comment by satwikkansal** [comment]
The web app is available at port 5000, not 8000. So this is expected behavior.

**Comment by satwikkansal** [comment]
Did you try running web app after this?

**Comment by parul-k** [comment]
> The web app is available at port 5000, not 8000. So this is expected behavior.

Above code is for NODE SERVER

**Comment by parul-k** [comment]
> Did you try running web app after this?

WEB APP is running fine 

![image](https://user-images.githubusercontent.com/60229741/77540004-59f58a80-6ec8-11ea-926d-e4403fd529fe.png)

Problem is of NODE SERVER

**Comment by satwikkansal** [comment]
Did you create some test transactions using the web app? What error did you get?

**Comment by parul-k** [comment]
> Did you create some test transactions using the web app? What error did you get?

Something worked finally,
Instead of writing http://127.0.0.1:8000/ which is saying Page Not Found, I wrote:
http://127.0.0.1:8000/mine
It said nothing to mine
http://127.0.0.1:5000/ opened and some text was entered
http://127.0.0.1:8000/mine displayed: Block #1 is mined.
After I clicked sync, it displayed my data..
This is how it was meant to be executed !!

**Comment by satwikkansal** [comment]
This is exactly what's written in the instructions here https://github.com/satwikkansal/python_blockchain_app#instructions-to-run

Anyways, closing the issue.

**Comment by parul-k** [comment]
> This is exactly what's written in the instructions here https://github.com/satwikkansal/python_blockchain_app#instructions-to-run
> 
> Anyways, closing the issue.

Thanks a lot for your kind support ..

**Comment by KS-Chain** [comment]
@parul-k  am facing the same problems as you did. I adapted everything from this feed ( I also need to mention, that am a pretty new in programming)
So my problem is still, that when pot 5000 opens and I type some information on the website, if i klick on Post or then also on request to mine I get: Not found (The request URL was not found on the server) - as you had the problem
I really don´t see or can find my mistake - I tried several things already.
Would be gread, if you or ohter could give me a hint. 
Thank you so much in adavance
